lisa/parameter_parser/runbook.py

```diff
@@ -125,8 +136,8 @@ def _merge_data(
     """
     result = data_from_parent.copy()
 
-    # merge others
-    result.update(data_from_current)
+    # Merge parent and current. We want a *deep/recursive* merge.
+    _update_dict_nested(result, data_from_current)
```

Comment:
It's by design here. To merge runbooks are very complex. Currently, it takes simplest way. Besides variable and parent paths, all other fields takes only highest level things. It's enough for current purpose. If you check current merge logic of variable and parent paths, there are a lot of tricky logic. For other parts, it's similar. Blinding deep merge will bring unexpected behavior. That's not easy to troubleshooting.
I thought to define a merge strategy like [this](https://pypi.org/project/jsonmerge/), it's not implemented yet. But it will be hard to use.

Reply:
If there is special logic on some part like variable, parent, I'm happy to have it. Because the merging is complex, currently the runbook should be composited by root level children, not support other scenarios. My principle is to provide a simple/limited feature, but not easy to make mistaken. Let me know, if it blocks something. We can figure out a better solution.

Reply:
But it is easy to make a mistake as it is. Write a root level entry on parent, same entry on current, you're toast: One of them gets ignored. Mine will only ignore up to *equal leaves*, if there are any.

Reply:
For merging runbooks, either approach cannot avoid all mistakes as far as I know. The current way takes latest, top one effective, and it should be completed. It's not recommended to define same child in different level. As what we used in `microsoft/runbook`, each level of runbook defines different nodes. the `tn.yml` define which cases to run, the `ready`, `local` and `azure` define the platform parameters.

Reply:
Alright, I'll can hold this to the future--I only have warn_as_error for now which I'd merge anyway. But this will re-surface again soon :/